code/events/psychopy2events.py

```diff
@@ -216,8 +216,121 @@ def pandas2bids(input_df: pd.DataFrame) -> pd.DataFrame:
         ][:len_remaining]
 
     df = df.replace({"value": {"nan": "n/a"}})
+    df = df.reset_index()
     return df[["onset", "duration", "trial_type", "value"]]
 
+def check_durations(events):
+    """ Test that the stimuli durations are close to the value they were set in the design of the tasks 
+
+    This function takes an input DataFrame with event information, runs various tests and 
+    raises exceptions if any test fails.
+    
+    Parameters
+    ----------
+    input_df : :obj:`pandas.DataFrame`
+        The input DataFrame containing BIDS-compatible event information.
+
+    """
+
+    EXPECTED_DURATION = {
+    "blank": 3,
+    "breath_in": 2.7,
+    "breath_in_last": 2.7,
+    "breath_out": 2.3,
+    "breath_out_last": 2.3,
+    "cog": 0.5,
+    "hold": 13,
+    "hold_end": 2,
+    "hold_test": 13,
+    "hold_test_end": 2,
+    "mot": 5,
+    "movie": 1200,
+    "vis": 3,
+    }
+    
+    for trial_type, expected_duration in EXPECTED_DURATION.items():
+        if trial_type in events["trial_type"].values:
+            indices = events[events["trial_type"] == trial_type].index
+            for index in indices:
+                duration = events.loc[index, "duration"]
+                if not abs(duration - expected_duration) < 0.2:
```

Comment:
Magic values should be defined at the head of the file, as constants:
```suggestion
                if not abs(duration - expected_duration) < EVENT_DURATION_EPSILON:
```

Reply:
(or better, use `numpy.isclose` with adequate tolerances)

Reply:
I tried, but I couldn't find a tolerance value adequate for the broad order of magnitude that we have to compare.
E.g. `cog` events should be around 0.5s (we need to accept 0.569 as well as 0.49s)
, `vis` events should be around 3s (we need to accept 3.17s)
while `movie` events are 1200s